caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/CadcResolver.java

```diff
@@ -0,0 +1,149 @@
+*                                       par écrit.
+*                                       
+*  This file is part of the             Ce fichier fait partie du projet
+*  OpenCADC project.                    OpenCADC.
+*                                       
+*  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+*  you can redistribute it and/or       vous pouvez le redistribuer ou le
+*  modify it under the terms of         modifier suivant les termes de
+*  the GNU Affero General Public        la “GNU Affero General Public
+*  License as published by the          License” telle que publiée
+*  Free Software Foundation,            par la Free Software Foundation
+*  either version 3 of the              : soit la version 3 de cette
+*  License, or (at your option)         licence, soit (à votre gré)
+*  any later version.                   toute version ultérieure.
+*                                       
+*  OpenCADC is distributed in the       OpenCADC est distribué
+*  hope that it will be useful,         dans l’espoir qu’il vous
+*  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+*  without even the implied             GARANTIE : sans même la garantie
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*                                       
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+*
+************************************************************************
+*/
+
+
+package ca.nrc.cadc.caom2.artifact.resolvers;
+
+import ca.nrc.cadc.auth.AuthMethod;
+import ca.nrc.cadc.auth.AuthenticationUtil;
+import ca.nrc.cadc.caom2.artifact.resolvers.util.ResolverUtil;
+import ca.nrc.cadc.net.StorageResolver;
+import ca.nrc.cadc.net.Traceable;
+import ca.nrc.cadc.reg.Standards;
+import ca.nrc.cadc.reg.client.RegistryClient;
+import org.apache.log4j.Logger;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+/**
+ * This class can convert an URI into a URL to the CADC storage system.
+ *
+ * @author adriand
+ */
+public class CadcResolver implements StorageResolver, Traceable {
+
+    private static final Logger log = Logger.getLogger(CadcResolver.class);
+    public String scheme = "cadc";
+    public static URI STORAGE_INVENTORY_URI;
+
```

Comment:
There are several things wrong here:
1. the variable should be `private` and `final`. 
2. `static` is certainly simple and one hard-codes a value... if it was not static, the value(s) would have to come from a config file, then multiple values could mean this works and files are accessible when some other services are down... TBD.
3. For a static URI, use `URI.create(String)` right in the declaration. You don't want to do error handling in a static block anyway and in this case if the string isn't a valid uri the unit tests won't be able to load this class and it will never get out of dev with that kind of bug.
For now, we'll use static, in which case the value has to be the global files service (ivo://cadc.nrc.ca/minoc is for local developer intTest usage only)... that doesn't exist and I tried to leave it up to ops to name stuff, but lets just put ivo://cadc.nrc.ca/global/files for now

Reply:
Sloppy on my part here. Fixed. Didn't know the `URI.create` trick - thanks